Optional Icon and Summary & Scrollable ListPreference and MultiSelectListPreference

NoelChew opened the pull request:
closes #6, closes #8

NoelChew committed:
Optional Icon and Summary

- allow nullable icon and summary values

NoelChew committed:
Scrollable ListPreference and MultiSelectListPreference

- added vertical scroll to Column modifier

NoelChew commented:
closes #8

NoelChew committed:
Set MultiSelectListPreference Dialog OK Button Text

- changed from hardcoded "Select" to stringResource(android.R.string.ok), which is supposedly localised

jollygreenegiant commented:
Any plans for this to be merged?  Would love to see these changes in the library

NoelChew commented:
While waiting for this to be merged, you can use this fork as a temporary dependency: 
implementation 'com.github.NoelChew.ComposePreferences:preferences-material:0.1.5-3'

NoelChew committed:
Material3 Version 1.0.1

- updated Jetpack Compose Material3 to v1.0.1
- set target SDK to 33
- added android:exported="true" to app AndroidManifest as required when targeting Android 12 and above

NoelChew committed:
Merge pull request #1 from NoelChew/material3-version-update

Material3 Version 1.0.1

NoelChew committed:
Update gradle-wrapper.properties

NoelChew committed:
Merge pull request #2 from NoelChew/material3-version-update

Update gradle-wrapper.properties

farmerbb committed:
Update dependencies

NoelChew committed:
Localised MultiSelectListPreferenceWidget Item Separator

- added localisation for the delimiter for user selected preference

NoelChew committed:
Merge pull request #3 from farmerbb/main

Update dependencies

NoelChew committed:
Merge pull request #4 from NoelChew/material3-version-update

Localised MultiSelectListPreferenceWidget Item Separator

NoelChew committed:
MultiSelectListPreferenceWidget Update

- allow user to set maximum number of selected value for display (optional)
- allow user to set custom summary text when all values are selected (optional)
- updated compileSdk and AGP version
- updated dependencies versions

NoelChew committed:
Merge pull request #5 from NoelChew/dev-20240704

MultiSelectListPreferenceWidget Update

NoelChew committed:
Java Version Update

NoelChew committed:
Merge pull request #6 from NoelChew/java-version-update

Java Version Update

NoelChew committed:
Update .jitpack.yml

NoelChew committed:
Merge pull request #7 from NoelChew/jetpack-yml-update

Update .jitpack.yml
